rcli/call.py

```diff
@@ -53,25 +53,25 @@ def call(func, args):
     """
     assert hasattr(func, '__call__'), 'Cannot call func: {}'.format(
         func.__name__)
-    is_func = isinstance(func, FunctionType)
-    raw_func = func if is_func else func.__class__.__call__
+    raw_func = (
+        func if isinstance(func, FunctionType) else func.__class__.__call__)
     hints = collections.defaultdict(lambda: Any, get_type_hints(raw_func))
-    params, vararg, kwarg = get_signature(raw_func)
-    params += [vararg, kwarg]
+    signature = getargspec(raw_func)
+    params = list(signature[0]) + [s for s in signature[1:3] if s]
     keyword_args = {}
     positional_args = ()
     for k, nk, v in _normalize(args):
-        if nk == vararg:
+        if nk == signature[1]:
             hints[nk] = Tuple[hints[nk], ...]
-        elif nk not in params and kwarg in hints:
-            hints[nk] = hints[kwarg]
+        elif nk not in params and signature[2] in hints:
+            hints[nk] = hints[signature[2]]
         try:
             value = cast(hints[nk], v)
-        except exc.CastError as e:
+        except TypeError as e:
             six.raise_from(exc.InvalidCliValueError(k, v), e)
-        if nk == vararg:
+        if nk == signature[1]:
             positional_args = value
-        elif (nk in params or kwarg) and (
+        elif (nk in params or signature[2]) and (
                 nk not in keyword_args or keyword_args[nk] is None):
             keyword_args[nk] = value
     return func(*positional_args, **keyword_args)
```

Comment:
I've made your requested changes, but instead of `pos_and_key_args` I've gone with `named_args`. Let me know if you disagree.